agent/src/agent/destination/repository.py

```diff
@@ -28,5 +28,16 @@ def delete():
     session().commit()
 
 
+def get_auth_token(destination_: HttpDestination) -> Optional[AuthenticationToken]:
+    if not destination_.id:
+        return None
+    return session().query(AuthenticationToken).filter(AuthenticationToken.destination_id == destination_.id).first()
```

Comment:
why don't you have a relationship in the destination model and get the token by `destination_.auth_token` or something like that?

Reply:
I just was implementing it :) You'll see it in the next commit